lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp

```diff
@@ -635,8 +635,29 @@ populateAIEv1CanonicalizeConversionPatterns(RewritePatternSet &patterns) {}
 static void
 populateAIEMLCanonicalizeConversionPatterns(RewritePatternSet &patterns) {}
 
+template <typename TransferOp> static void setInBounds(TransferOp op) {
+  if (op.getTransferRank() == 0)
+    return;
+  SmallVector<bool, 4> bools(op.getTransferRank(), true);
+  OpBuilder b(op.getContext());
+  op->setAttr(op.getInBoundsAttrName(), b.getBoolArrayAttr(bools));
+}
+
 void CanonicalizeForAIEVecPass::runOnOperation() {
   func::FuncOp funcOp = getOperation();
+
+  funcOp.walk([&](Operation *Op) {
+    if (auto readOp = dyn_cast<TransferReadOp>(Op)) {
+      if (!readOp.getInBounds())
+        setInBounds<TransferReadOp>(readOp);
+    } else if (auto writeOp = dyn_cast<TransferWriteOp>(Op)) {
+      if (!writeOp.getInBounds())
+        setInBounds<TransferWriteOp>(writeOp);
+    }
+  });
```

Comment:
Navigating the code ourselves and making in-place modifications leads to bad failure modes and potentially incorrect transformations. If we need to do this, it should be done in its own "TransferReadOp"-bound rewrite/conversion. For something of this nature (marking transfer_reads as in-bound), a proper pre-analysis is probably the right approach.